### DISCUSSION TITLE
wg-ui-plus running under podman.

### DISCUSSION THREAD

**Discussion by tetricky** [vijaygill/wg-ui-plus#155] [Show and tell]
Just wanted to say thanks for this project. It very much hits a sweet spot for one stop vpn server configuration. It makes it very simple to group peers together with configuration and endpoints to achieve network separation and meet routing requirements.

I'm running on a cloud server behind authelia (to restrict access) with caddy as a reverse-proxy, and running in a podman pod and container.

First I create the pod, with port mapping:
```
podman pod create -p "51820:51820/udp" -p "8000:8000" --name vpn
```
Creating the directories I use for persistent data is a simple:
```
mkdir -p /var/lib/containers/userland/wg-ui-plus/{config,data}
chmod og+w /var/lib/containers/userland/wg-ui-plus/{config,data}
```

I also pull the image and tag locally, while using `--label io.containers.autoupdate=local` to manage auto-updates, but the main functional container command is a relatively simple:

```
podman run --pod=vpn -d --name wguiplus \
 -v /lib/modules:/lib/modules:ro \
 -v /tmp:/tmp \
 -v /var/lib/containers/userland/wg-ui-plus/config:/config \
  -v /var/lib/containers/userland/wg-ui-plus/data:/data \
 --cap-add=NET_ADMIN \
 --cap-add SYS_MODULE \
 --sysctl net.ipv4.conf.all.src_valid_mark=1 \
 --sysctl net.ipv4.ip_forward=1 \
 --cap-add=NET_RAW \
  ghcr.io/vijaygill/wg-ui-plus:latest-v1.2.7
```
That's it. Works perfectly. Caddy is set to proxy at the servers ip address and port 8000, and authelia ensures that only correctly authenticated users can access the portal.

I wanted to articulate my thanks, and provide a podman example in order to encourage others.
Thanks!

**Comment by vijaygill** [comment]
@tetricky - Thank you so much for posting the feedback and also the use-case that might suit many others.
Looking at your post, I think I should have named targets as "End Points". But naming things has never been strong point.
